File: experiments/esbmc_limit_sweep/cases/02_loop/bad.c
#include <assert.h>

int nondet_int(void);

int f(int x, int n)
{
    int acc = 0;
    int i = 0;

    while (i < n)
    {
        /* MUTATION */
        acc = acc + x + i + 1;
        i = i + 1;
    }

    return acc;
}

int main(void)
{
    int x = nondet_int();
    int n = nondet_int();

    __ESBMC_assume(0 <= x);
    __ESBMC_assume(x <= 20);

    __ESBMC_assume(0 <= n);
    __ESBMC_assume(n <= 5);

    int result = f(x, n);

    int expected =
        n * x + (n * (n - 1)) / 2;

    assert(result == expected);

    return 0;
}

File: experiments/esbmc_limit_sweep/phase3a_cases/10_regex/bad.py
import re

k: int = nondet_int()
__ESBMC_assume(0 <= k <= 2)

s: str = "abc"

if k == 0:
    s = "abc"
elif k == 1:
    s = "a1"
else:
    s = "xyz"

matched: bool = re.fullmatch("[0-9]+", s) is not None

expected: bool = False

if k == 0:
    expected = True
elif k == 2:
    expected = True

assert matched == expected
